api_spec.yaml

```diff
@@ -2847,6 +2825,34 @@ paths:
         '500':
           $ref: '#/components/responses/500'
 
+  /.well-known/jwks.json:
+    get:
+      tags:
+        - JWT Recipe
+      description: |
+        Retrieve JWKs for JWT verification
```

Comment:
might want to also elaborate that this contains static and dynamic keys + how will the dev know which keys in the list are static vs dynamic ones?

Reply:
Why would they need to know? If they create a JWT/access token and want to find the appropriate key, they can look for the key id.
In any case, if they really need to know, then the ids will be in a different format - UUID vs just numbers. This is not a great way to distinguish them, but the response to this endpoint needs to match the standards.

Reply:
Right. The reason they would need to know is if someone wants to hard code the public key in their server's env. So in our docs, we say that they should query the jwks endpoint and find the keys from there - but now some of them are dynamic - so they would need to know which one to pick. 

As long as it's explainable in the docs, thats fine. I was thinking that the first key displayed in the list is always the static one.

About the id format - we should probably stick to what the standard says and keep the format consistent across the static and dynamic keys? Not sure.

Reply:
We can list them and make them ordered like that, but one issue is that we only generate a static key on demand - meaning the first (or only) key could be a dynamic one if they never created a token signed by static keys.

The best way to get the appropriate static key (and to ensure one is created) is to create a token and check the key id. Of course, we could return this info in the token-creating responses, but it can also be parsed from the token. (i.e.: pasting it into an online JWT parser)

Reply:
> but one issue is that we only generate a static key on demand

We could change this and create a static key during core start - if it doesn't already exist (this is what we do now anyway)

> The best way to get the appropriate static key (and to ensure one is created) is to create a token and check the key id

Agreed. But this is complex to communicate, and is prone to mistakes. Anyway, we can recommend this method - sure.

Reply:
We decided to add `s-` and `d-` as a prefix.